grpc/server/test/connection_initiator_test.cpp

```diff
@@ -0,0 +1,84 @@
+#include <future>
+#include <gtest/gtest.h>
+
+#include "connection_initiator.h"
+#include "mocks/dronecore_mock.h"
+
+using testing::_;
+using testing::NiceMock;
+
+using event_callback_t = dronecore::testing::event_callback_t;
+using MockDroneCore = NiceMock<dronecore::testing::MockDroneCore>;
+using ConnectionInitiator = dronecore::backend::ConnectionInitiator<MockDroneCore>;
+
+static constexpr auto ARBITRARY_PORT = 1291;
+static constexpr auto ARBITRARY_UUID = 1492;
+
+ACTION_P(SaveCallback, event_callback)
+{
+    *event_callback = arg0;
+}
+
+TEST(ConnectionInitiator, registerDiscoverIsCalledExactlyOnce)
+{
+    ConnectionInitiator initiator;
+    MockDroneCore dc;
+    EXPECT_CALL(dc, register_on_discover(_))
+    .Times(1);
+
+    initiator.start(dc, ARBITRARY_PORT);
+}
+
+TEST(ConnectionInitiator, startAddsUDPConnection)
+{
+    ConnectionInitiator initiator;
+    MockDroneCore dc;
+
+    EXPECT_CALL(dc, add_udp_connection(_));
+
+    initiator.start(dc, ARBITRARY_PORT);
+}
+
+TEST(ConnectionInitiator, startHangsUntilDeviceDiscovered)
+{
+    ConnectionInitiator initiator;
+    MockDroneCore dc;
+    event_callback_t discover_callback;
+    EXPECT_CALL(dc, register_on_discover(_))
+    .WillOnce(SaveCallback(&discover_callback));
+
+    auto async_future = std::async(std::launch::async, [&initiator, &dc]() {
+        initiator.start(dc, ARBITRARY_PORT);
+        initiator.wait();
+    });
+
+    EXPECT_TRUE(async_future.wait_for(std::chrono::milliseconds(100)) == std::future_status::timeout) <<
```

Comment:
I usually use fake time for unit tests. Have you seen that?

Reply:
Oh, I've seen a `FAKE_TIME`, but immediately forgot about it. I'll have a look!

Reply:
I am not sure how I should use it here, actually. In `call_every_handler_test.cpp`, I see the benefit of the fake time because the object under test is using the fake time as well. But here, I don't know how I can make my `initiator.wait()` use the fake time?

I am a bit confused, maybe that's because I am not used to `FakeTime` ^^. How would you do it?